src/QueueProvider.php

```diff
@@ -38,11 +37,8 @@ public function getChannel(): AMQPChannel
         if ($this->channel === null) {
             $this->channel = $this->connection->channel();
             $this->channel->queue_declare(...$this->queueSettings->getPositionalSettings());
-
-            if ($this->exchangeSettings !== null) {
-                $this->channel->exchange_declare(...$this->exchangeSettings->getPositionalSettings());
-                $this->channel->queue_bind($this->queueSettings->getName(), $this->exchangeSettings->getName());
-            }
+            $this->channel->exchange_declare(...$this->exchangeSettings->getPositionalSettings());
```

Comment:
`$this->exchangeSettings` can be null, so an error may occur in this line. Refactor class to exclude case when `private ?AMQPChannel $channel = null;` may be null.